tgui/packages/tgui/interfaces/Hypertorus.js

```diff
@@ -0,0 +1,302 @@
+import { Window } from '../layouts';
+import { formatSiUnit, formatSiBaseTenUnit } from '../format';
+
+export const Hypertorus = (props, context) => {
+  const { act, data } = useBackend(context);
+  const filterTypes = data.filter_types || [];
+  const {
+    energy_level,
+    core_temperature,
+    internal_power,
+    power_output,
+    heat_limiter_modifier,
+    heat_output,
+    heat_output_bool,
+    heating_conductor,
+    magnetic_constrictor,
+    fuel_injection_rate,
+    moderator_injection_rate,
+    current_damper,
+    power_level,
+    iron_content,
+    integrity,
+    start_power,
+    start_cooling,
+    start_fuel,
+    internal_fusion_temperature,
+    moderator_internal_temperature,
+    internal_output_temperature,
+    internal_coolant_temperature,
+    waste_remove,
+  } = data;
+  const fusion_gases = flow([
+    filter(gas => gas.amount >= 0.01),
+    sortBy(gas => -gas.amount),
+  ])(data.fusion_gases || []);
+  const moderator_gases = flow([
+    filter(gas => gas.amount >= 0.01),
+    sortBy(gas => -gas.amount),
+  ])(data.moderator_gases || []);
+  const fusionMax = Math.max(1, ...fusion_gases.map(gas => gas.amount));
+  const moderatorMax = Math.max(1, ...moderator_gases.map(gas => gas.amount));
+  return (
+    <Window
+      width={500}
+      height={600}
+      scrollable
+      resizable
+      title="Fusion Reactor">
+      <Window.Content>
+        <Section title="Switches">
+          <Flex m={-0.5}>
+            <Flex.Item m={0.5} color="label">
+              {'Start power: '}
+              <Button
+                disabled={data.power_level > 0}
+                icon={data.start_power ? 'power-off' : 'times'}
+                content={data.start_power ? 'On' : 'Off'}
+                selected={data.start_power}
+                onClick={() => act('start_power')} />
+            </Flex.Item>
+            <Flex.Item m={0.5} color="label">
+              {'Start cooling: '}
+              <Button
+                disabled={start_fuel === 1
+                    || start_power === 0
+                    || data.power_level > 0}
+                icon={data.start_cooling ? 'power-off' : 'times'}
+                content={data.start_cooling ? 'On' : 'Off'}
+                selected={data.start_cooling}
+                onClick={() => act('start_cooling')} />
+            </Flex.Item>
+            <Flex.Item m={0.5} color="label">
+              {'Start fuel injection: '}
+              <Button
+                disabled={start_power === 0
+                    || start_cooling === 0}
+                icon={data.start_fuel ? 'power-off' : 'times'}
+                content={data.start_fuel ? 'On' : 'Off'}
+                selected={data.start_fuel}
+                onClick={() => act('start_fuel')} />
+            </Flex.Item>
+          </Flex>
+        </Section>
+        <Section>
+          <Section title="Internal Fusion Gases">
+            {fusion_gases.map(gas => (
+              <LabeledList.Item
```

Comment:
**`LabeledList` structure violation.**

`LabeledList.Item` elements must be direct children of `LabeledList`. It's like having table rows without a table element. It also breaks styling, padding without it is incorrect.